src/smc-webapp/buttonbar.coffee

```diff
@@ -51,11 +51,280 @@ exports.commands =
                 right : ''
         set_name_and_email :
                 insert:
-                         """
-                         git config --global user.name ""
-                         git config --global user.email ""
-                         """
+                        """
+                        git config --global user.name ""
+                        git config --global user.email ""
+                        """
+        initalize_git :
+                insert: 
+                        """
+                        git init
+                        """  
+        create_gitignore :
+                insert:
```

Comment:
for me, this gitignore is way too large. please reduce it to the minimals, like `*.py[cod]` etc.